Jetcaster/app/src/main/java/com/example/jetcaster/ui/JetcasterAppState.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.jetcaster.ui
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Build
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.platform.LocalContext
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.rememberNavController
+
+/**
+ * List of screens for [JetcasterApp]
+ */
+sealed class Screen(val route: String) {
+    object Home : Screen("home")
+    object Player : Screen("player/{podcastUri}") {
+        fun createRoute(podcastUri: String) = "player/$podcastUri"
+    }
+}
+
+@Composable
+fun rememberJetcasterAppState(
+    navController: NavHostController = rememberNavController(),
+    context: Context = LocalContext.current
+) = remember(navController, context) {
+    JetcasterAppState(navController, context)
+}
+
+class JetcasterAppState(
+    val navController: NavHostController,
+    private val context: Context
+) {
+    var isOnline by mutableStateOf(checkIfOnline())
+        private set
+
+    fun refreshOnline() {
+        isOnline = checkIfOnline()
+    }
+
+    @Suppress("DEPRECATION")
+    private fun checkIfOnline(): Boolean {
+        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
```

Comment:
Small nit: I like the `androidx.core` extension here, which avoids having to know the right `Context` constant:
```suggestion
        val cm = context.getSystemService<ConnectivityManager>()!!
```

Reply:
But let's keep it nullable.

Reply:
Done! Thanks